src/main/kotlin/net/maizegenetics/phgv2/pathing/DiploidPathFinding.kt

```diff
@@ -0,0 +1,336 @@
+
+        //map of sampleName -> list of read mapping file names
+        val sampleToFiles = keyFileLines.drop(1).associate { Pair(it[0], it[1]) }
+            .mapValues { it.value.split(",").map{filename -> filename.trim()} }
+
+        return checkForExistingOutput(sampleToFiles)
+    }
+
+    /**
+     * Takes a map of samples to file list as input. If the output directory contains an hvcf file for any sample,
+     * that sample will be removed from the map and the trimmed map will be returned.
+     */
+    private fun checkForExistingOutput(sampleToFiles: Map<String, List<String>>): Map<String, List<String>> {
+        val existingSampleFiles = File(outputDir).listFiles()
+        val existingSamples = existingSampleFiles.map {file ->
+            file.name.substringBefore(".h.vcf", "$$$")
+        }.filter { it != "$$$" }
+
+        val newSamples = sampleToFiles.filter { (sampleName, _) -> !existingSamples.contains(sampleName) }
+
+        if (newSamples.size < sampleToFiles.size)
+            myLogger.info("The key file $pathKeyfile contains ${sampleToFiles.size}, only ${newSamples.size} will be imputed." +
+                    "The others already existing in the output directory")
+
+        return newSamples
+    }
+
+    private data class ReadMappingResult(val name: String, val readMappingCounts: Map<List<String>, Int>) //placeholder
+//    private data class Path(val name: String, val hapidList: List<List<String>>, val graph: HaplotypeGraph, val likelyParents: List<MostLikelyParents.ParentStats>)
+    private data class Path(val name: String, val hapidList: List<PathFinderWithViterbiHMM.DiploidPathNode>, val graph: HaplotypeGraph, val likelyParents: List<MostLikelyParents.ParentStats>)
+    /**
+     * Takes a map of sample name -> list of read mapping files. It processes the samples in parallel.
+     * Multi-threading is across samples. The results are written to files by an additional thread.
+     * The results are a h.vcf for each sample and, if the most likely parents are used, a single file of
+     * the chosen ancestors (parents) for each sample.
+     */
+    private fun processReadMappings(sampleToFiles: Map<String, List<String>>) = runBlocking {
+        myLogger.info("processing read mappings.")
+        val myGraph = buildHaplotypeGraph()
+
+        //check onditions for useLikelyAncestor
+        if (useLikelyAncestors) {
+            if(maxAncestors >= myGraph.sampleGametesInGraph().size && minCoverage == 1.0)
+                myLogger.warn("UseLikelyAncestors is true but likelyAncestors will not be checked because minCoverage is 1.0 " +
+                        "and maxAncestors is >= the number of potential ancestors.")
+            else if (likelyAncestorFile.isBlank()) myLogger.warn("Likely ancestors will be determined for each sample " +
+                    "but information about the ancestors used will not be written to a file because no file name was provided.")
+        }
```

Comment:
Do we just want to warn here, or throw an error and exit? If we just warn then a user won't notice the message and will wonder why they didn't get the desired results.